test/integration/dataplane_mtls_test.go

```diff
@@ -0,0 +1,118 @@
+//go:build dataplane_tests
+// +build dataplane_tests
+
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package integration
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+var (
+	CAFile         = certFile("ca.pem")
+	ClientCertFile = certFile("client.pem")
+	ClientKeyFile  = certFile("client-key.pem")
+	ServerCertFile = certFile("server.pem")
+	ServerKeyFile  = certFile("server-key.pem")
+)
+
+// TestGRPCClient tests the gRPC client against a running Docker container.
+func TestGRPCClient(t *testing.T) {
```

Comment:
Is it worth pointing out here (for the future code reader) that this test is verifying our mTLS configuration works?

```suggestion
// TestGRPCClientWithMTLS tests the client connectivity to the dataplane's gRPC API server. Mutual TLS (mTLS)
// is required to connect to the dataplane API server, and so as a part of this test we create a client certificate
// using the same CA the server used in order to establish mutual trust.
func TestGRPCClientWithMTLS(t *testing.T) {
```

Maybe a bit too wordy, so feel free to ignore this specific suggestion, but I think a breadcrumb here that let's future readers better understand that mTLS configuration is effectively being tested here would be good.